### ISSUE TITLE
Not Sure How To Configure Roombapw.

### ISSUE THREAD

**Issue by Penguin-Guru** [jeremywillans/ha-rest980-roomba#14]
I'm trying to install for Hassio and probably made one or more mistakes along the way. I *think* everything is installed but nothing has appeared to suggest it's working. One obvious issue I ran into is Step 1 of the "how to use" for Roombapw. Where am I supposed to "set the required ROBOT_ID configuration attribute"? No dialogues offered me any options for that and there is no mention of any config files. Sorry if this is a stupid question, I'm just starting to learn about Home Assistant and have never used Docker before.

I do already have the robots' connection information, since I was able to set this up previously: https://www.home-assistant.io/integrations/roomba/#retrieving-your-credentials

Also, your documentation references Docker/NGINX hosts and ports-- where are these configured, or at least identified?

**Comment by jeremywillans** [comment]
Ok if you have got the roomba creds, then you dont need RoombaPW (as its only used to get the creds!)

Essentially when you add a new addon in HA (formerly HASS) there is a configuration section which allows you to set variables. this is where you define both the ROBOT_IP address in the case of RoombaPW, as well as where the ports are defined for the rest980 (default 3000) and php_nginx (default 3001) images.

Hope this helps!

**Comment by Penguin-Guru** [comment]
Thanks, I found the Configuration tab! Still getting used to this interface it seems...

Could you confirm whether "vacuum.yaml" and "lovelace.yaml" should be created in "/config"? That's my assumption, since it's where "secrets.yaml" is, but I'm still having some trouble.

The trouble I'm having now is: nothing appeared in my Overview tab but the add-on logs both seem normal. Opening image.php in a web browser shows mostly binary data but here are some highlights.

At the beginning:

> Warning: file_get_contents(http://127.0.0.1:3001/vacuum.log?v=1606182058): failed to open stream: Connection refused in /config/vacuum/image.php on line 50

> Notice: Undefined offset: -1 in /config/vacuum/image.php on line 57

> Warning: Cannot modify header information - headers already sent by (output started at /config/vacuum/image.php:50) in /config/vacuum/image.php on line 241


At the end:

> Notice: Undefined variable: roomba_stuck in /config/vacuum/image.php on line 249

> Warning: imagedestroy() expects parameter 1 to be resource, null given in /config/vacuum/image.php on line 249

> Warning: imagedestroy(): supplied resource is not a valid Image resource in /config/vacuum/image.php on line 250

There is no such log file in the vacuum directory so I tried touching one and restarting but that didn't help. It is supposed to be the nginx port, right?

**Comment by jeremywillans** [comment]
This deployment uses the HA Packages feature so you dont need to copy the content of vacuum.yaml into any other files. The lovelace file content you need to copy into two lovelace cards, likely through the UI.

You cant use IP Address 127.0.0.1  -> Replace with the actual IP Address of your HA Box (as this is accessed from the browser of your computer, its trying to connect to itseelf which is why your getting the error.

**Comment by jeremywillans** [comment]
Bump

**Comment by Penguin-Guru** [comment]
Sorry for the delay. I got sidetracked by some other HA issues but that's no excuse...

I tried changing the IP address to the system's actual address but the output from image.php looks the same to me. The messages I mentioned before are all still present. I tried restarting HA and even restarting the whole system. Is there something I need to do to generate new content for image.php?

I have not copied the contents of "vacuum.yaml", I just installed it in "/config". I'm not sure how to configure lovelace but will try to figure that out and see if that's what's causing these messages.

Is there supposed to be a "vacuum.log" file in "/config/vacuum/" ? I see that's defined in "secret.yaml" and I thought it would create itself automatically but it doesn't seem to have done that. Perhaps because no connection with the robot has been successful?

**Comment by Penguin-Guru** [comment]
I tried adding the cards through the UI like you said (after deleting the old lovelace.yaml and restarting HA). I pasted in the card code for just the first item and fixing the spacing, but it said "UI editor is not supported for this config:" "Cannot read property 'setConfig' of undefined". I then tried adding just the second card item, but the same error appeared. Adding the first and second together was the same. Do you know where this call for 'setConfig' is coming from?

**Comment by jeremywillans** [comment]
>  have not copied the contents of "vacuum.yaml", I just installed it in "/config"

this isnt going to work. the vacuum.yaml file is intended to be used with packages component of HA

for simplicity - 
in your config directory, create a new folder called packages and put the vacuum.yaml in this folder.
edit configuration.yaml add the following (note: if there is already a `homeassistant:` line in your file, then just add the packages line as an additional entry in this section)
```
homeassistant:
  packages: !include_dir_named packages
```

without this, none of the required sensors are created resulting in the other errors you are getting.

**Comment by Penguin-Guru** [comment]
Ok, thank you! I haven't tested everything yet but this does seem to be working now. I'm still not sure what the packages component is but I don't want to waste your time explaining that. I'll look into it. :)

There are a few other things I did that don't seem to be mentioned in the instructions, although they probably should have been obvious if I knew what I was doing:

- Edited "vacuum.yaml" to use the areas I defined in "secret.yaml".
- Edited the U.I. card to use the areas I defined in "secret.yaml".
- Commented out vacuum_notify sections in "vacuum.yaml".
